docs/storage.md

```diff
@@ -137,19 +137,16 @@ To learn more about existing integrations with remote storage systems, see the [
 If a user wants to create blocks into the TSDB from data which is in [OpenMetrics](https://openmetrics.io/) format, they can do so using backfilling. However, they should be careful and note that it is not safe to backfill data from the last 3 hours (the current head block).
 
 
-The backfill sub-command is implemented such that it keeps at most 2 hours of data in the memory. Hence, it is important to make sure that there is enough RAM for 2 hours worth of data at a time.
+The backfill command is implemented such that Prometheus keeps at most 2 hours of data in the memory. 
 
 An example use case of backfilling is :
-If a user wants to migrate from an existing monitoring system (source) to Prometheus, they can add their older data to TSDB easily using backfilling. The source system may be another TSDB which is capable of dumping data in [OpenMetrics](https://openmetrics.io/) format. 
+If a user wants to migrate from an existing monitoring system (source) to Prometheus, they can add their older data to TSDB using backfilling if their data is in [OpenMetrics](https://openmetrics.io/) format. 
 
 ### Usage 
 
 Backfilling can be used via the promtool command line. Promtool will write the blocks to a directory. By default this output directory is data/, you can change it by using the name of the desired output directory as an optional argument in the sub-command.
 
 ```
-./promtool create-blocks-from openmetrics name_of_input_file name_of_output_directory
+./promtool create-blocks-from openmetrics name_of_input_file [name_of_output_directory]
 ```
-
-If there is an overlap, you will need to shut down Prometheus, and restart it with `--storage.tsdb.allow-overlapping-blocks`. If there is no overlap or if the flag is already set, no shut-down is required. 
-
-If you do not want to lose any importannt data points, use the `--storage.tsdb.retention.time=X` flag. 
+After the creation of the blocks, move it to the desired directory. If there is an overlap with older data, the flag `--storage.tsdb.allow-overlapping-blocks` needs to be set. Also, if `--storage.tsdb.retention.time=X` flag is not set the data older than the retention will be deleted.
```

Comment:
Desired is a bit vague, maybe can move it to the data directory of Prometheus?

s/older data/existing blocks in Prometheus.

>  Also, if `--storage.tsdb.retention.time=X` flag is not set the data older than the retention will be deleted.

This is incorrect. It's more that be aware that if time retention is configured, that it may affect your blocks.

Reply:
I think you should mention that `--storage.tsdb.allow-overlapping-blocks` is a Prometheus flag (not a `promtool` one).

And what Brian said.

Suggestion for the second sentence: "Note that any backfilled data is subject to the retention configured for your Prometheus server (by time or size)."

Reply:
Actual question about the content: Shouldn't it just work to backfill directly into the Prometheus data directory, even for a live Prometheus server? There is the risk of a partial backfill not being able to revert etc., but are there technical problems?

Reply:
Yes there is a risk.

When you create a block, it creates a .tmp-for-creation directory.

If at the same time Prometheus runs a compaction, it would delete that block which is being backfilled.

Reply:
From discussions the intention was to have a separate subcommand to do the moving with safety checks. Personally I think it should be an all-in-one tool to make things easy for users, but that's not where the discussion went.

Reply:
We don't have to solve this now. I was just thinking it should "just work". But if we want an explicit "validate and move" step in the command, also fine. Let's do whatever turns out to be the best fit for the problem. But we don't have it now, so let's document what is documented now (create separately and move).

> If at the same time Prometheus runs a compaction, it would delete that block which is being backfilled.

Is the compaction really deleting all tmp directories? We do have all those UUIDs, so the compaction could restrict itself to only delete tmp directories it is in charge of.

Reply:
Let's continue this discussion separately. I'll resolve this thread.